MessagingService/src/test/java/CS4337/Project/MessagingServiceTest.java

```diff
@@ -0,0 +1,182 @@
+
+        assertEquals(HttpStatus.INTERNAL_SERVER_ERROR, response.getStatusCode());
+        assertEquals("Database error: Database error", response.getBody().get("error"));
+    }
+
+    @Test
+    public void testRemoveChatSuccess() {
+        int chatId = 1;
+        when(chatParticipantRepository.delChatParticipant(chatId)).thenReturn(1);
+
+        ResponseEntity<Map<String, Object>> response = messagingService.removeChat(chatId);
+
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertEquals("Chat participant removed successfully.", response.getBody().get("message"));
+    }
+
+    @Test
+    public void testRemoveChatNotFound() {
+        int chatId = 1;
+        when(chatParticipantRepository.delChatParticipant(chatId)).thenReturn(0);
+
+        ResponseEntity<Map<String, Object>> response = messagingService.removeChat(chatId);
+
+        assertEquals(HttpStatus.NOT_FOUND, response.getStatusCode());
+        assertEquals("Chat participant not found.", response.getBody().get("error"));
+    }
+
+    @Test
+public void testGetChatParticipantsUnexpectedException() {
+    // Arrange
+    when(chatParticipantRepository.getAllChatParticipant()).thenThrow(new RuntimeException("Unexpected error"));
+
+    // Act
+    ResponseEntity<Map<String, Object>> response = messagingService.getChatParticipants();
+
+    // Assert
+    assertEquals(HttpStatus.INTERNAL_SERVER_ERROR, response.getStatusCode());
+    assertEquals("Unexpected error occurred.", response.getBody().get("error"));
+}
+
+@Test
+public void testGetChatParticipantsWhenNoneExist() {
+    // Arrange
+    when(chatParticipantRepository.getAllChatParticipant()).thenReturn(Collections.emptyList());
+
+    // Act
+    ResponseEntity<Map<String, Object>> response = messagingService.getChatParticipants();
+
+    // Assert
+    assertEquals(HttpStatus.OK, response.getStatusCode());
+    assertEquals(Collections.emptyList(), response.getBody().get("success"));
+}
+
+@Test
+public void testRemoveChatWhenNoParticipants() {
+    // Arrange
+    int chatId = 1;
+    when(chatParticipantRepository.delChatParticipant(chatId)).thenReturn(0); // No participants to remove
+
+    // Act
+    ResponseEntity<Map<String, Object>> response = messagingService.removeChat(chatId);
+
+    // Assert
+    assertEquals(HttpStatus.NOT_FOUND, response.getStatusCode());
+    assertEquals("Chat participant not found.", response.getBody().get("error"));
+}
+
+@Test
+public void testSendMessageInvalidChatId() {
+    // Arrange
+    Map<String, Object> messageDetails = new HashMap<>();
+    messageDetails.put("chatid", -1); // Invalid chat ID
+    messageDetails.put("senderid", 1);
+    messageDetails.put("content", "Hello!");
+
+    // Act
+    ResponseEntity<Map<String, Object>> response = messagingService.sendMessage(messageDetails);
+
+    // Assert
+    assertEquals(HttpStatus.BAD_REQUEST, response.getStatusCode());
+    assertEquals("Invalid chat ID.", response.getBody().get("error"));
```

Comment:
The response body is "Chat ID and Sender ID cannot be null." instead of "Invalid chat ID".